src/screens/SendSheet.js

```diff
@@ -395,22 +388,38 @@ export default function SendSheet(props) {
       });
       analytics.track('Changed native currency input in Send flow');
     },
-    [maxInputBalance, selected.decimals, selected?.price?.value]
+    [maxEnabled, maxInputBalance, selected.decimals, selected?.price?.value]
   );
 
-  const sendMaxBalance = useCallback(async () => {
-    const newBalanceAmount = await updateMaxInputBalance(selected);
-    sendUpdateAssetAmount(newBalanceAmount);
-  }, [selected, sendUpdateAssetAmount, updateMaxInputBalance]);
+  useEffect(() => {
+    const updateMaxBalance = async () => {
```

Comment:
forgot to mention one more thing - this was a pre-existing issue but `updateMaxInputBalance` actually isn't an async function, so you can remove the `await` and just simply do these updates if `maxEnabled`